runtime/layers/console/bootstrap

```diff
@@ -51,12 +51,26 @@ while (true) {
             $cliOptions
         );
 
-        exec($command, $output, $exitCode);
+        $descriptors = [
+            0 => ['pipe', 'r'], // stdin, unused
+            1 => ['pipe', 'w'], // stdout
+            2 => ['pipe', 'w'], // stderr, unused
+        ];
+
+        $process = proc_open($command, $descriptors, $pipes);
+        $output = '';
+
+        do {
+            $status = proc_get_status($process);
 
-        $output = implode("\n", $output);
+            while (false !== $data = fgets($pipes[1])) {
+                echo $data;
+                $output .= $data;
+            }
+        } while ($status['running']);
```

Comment:
Thanks for the PR!

This will run the process in a loop, which will consume a lot of CPU resources, is that correct?

Reply:
Thanks for the speedy response.

Good point, as it stands this would just loop and consume resources.

There may be something we can do with `stream_select()` to wait a short time for changes on stdout before going round the loop again (potentially based on `deadlineMs`). Or alternatively we may be able to cause `fgets()` to block with `socket_set_blocking()`, which would mean we don't loop until there is data to read (or the process ends, need to experiment).

I can experiment with these ideas and come back to you.

Reply:
Thanks for the details!

Maybe it could be worth looking into the Symfony Process component? I believe it has the ability to do that already, and we already require this package in composer.json.

Reply:
Great idea! It looks like the Process component makes this a lot simpler. Would be good to get your thoughts on this. Cheers